includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -1292,6 +1292,10 @@ public function process_payment( $order_id ) {
 			$payload     = array();
 
 			if ( $doing_classic_payment ) {
+				if ( empty( $order->get_shipping_phone() ) && empty( $order->get_billing_phone() ) ) {
```

Comment:
It would be possible, but if we remove the validation from there we are depending only on front-end validation when the merchant uses WooCommerce blocks. woocommerce_after_checkout_validation action does not run when blocks checkout is used.

I'd suggest better to have the check once for both cases.

Reply:
When we have it on the process_payment method. We have backend validation in blocks checkout or shortcode checkout.
In the blocks case, we have implemented a pre-backend validation that shows nicely, but even if that gets overwritten and a backend call takes place, an error will be displayed because of the caught exception.
In the shortcode case, simply the Exception is being caught and its message being displayed as a wc_add_notice.

Reply:
Still think we should a validation check in `woocommerce_after_checkout_validation` for the non-blocks classic flow. 
It would avoid creating the order as validation should.